extensions/azure-extensions/src/main/java/io/druid/storage/azure/AzureDataSegmentPusher.java

```diff
@@ -100,12 +99,15 @@ public File createSegmentDescriptorFile(final ObjectMapper jsonMapper, final Dat
     return RetryUtils.retry(f, AzureUtils.AZURE_RETRY, maxTries);
   }
 
-  public void uploadThenDelete(final File file, final String azurePath)
+  public long uploadThenDelete(final File file, final String azurePath)
```

Comment:
You should probably do the cleanup (delete) as part of the push method, not here. If this fails for whatever reason, then you can partially clobber your data and any subsequent calls to the azure retry should fail because the data has already been deleted.